src/next-transpile-modules.js

```diff
@@ -143,13 +81,68 @@ const withTmInitializer = (modules = [], options = {}) => {
   const withTM = (nextConfig = {}) => {
     if (modules.length === 0) return nextConfig;
 
+    // IMPROVE ME: false should not be the default for this option
     const resolveSymlinks = options.resolveSymlinks || false;
     const isWebpack5 = (nextConfig.future && nextConfig.future.webpack5) || false;
     const debug = options.debug || false;
 
     const logger = createLogger(debug);
 
-    const modulesPaths = generateModulesPaths(modules);
+    /**
+     * Our own Node.js resolver that can ignore symlinks resolution and  can support
+     * PnP
+     */
+    const resolve = enhancedResolve.create.sync({
+      symlinks: resolveSymlinks,
+      extensions: ['.js', '.jsx', '.ts', '.tsx', '.mjs', '.css', '.scss', '.sass'],
+      mainFields: ['main', 'module', 'source'],
+      // Is it right? https://github.com/webpack/enhanced-resolve/issues/283#issuecomment-775162497
+      conditionNames: ['require'],
+    });
+
+    /**
+     * Return the root path (package.json directory) of a given module
+     * @param {string} module
+     * @returns {string}
+     */
+    const getPackageRootDirectory = (module) => {
+      let packageDirectory;
+      let packageRootDirectory;
+
+      try {
+        // Get the module path
+        packageDirectory = resolve(CWD, module);
+
+        if (!packageDirectory) {
+          throw new Error(
+            `next-transpile-modules - could not resolve module "${module}". Are you sure the name of the module you are trying to transpile is correct?`
+          );
+        }
+
+        // Get the location of its package.json
+        const pkgPath = escalade(packageDirectory, (dir, names) => {
+          if (names.includes('package.json')) {
+            return 'package.json';
+          }
+          return false;
+        });
+        if (pkgPath == null) {
+          throw new Error(
+            `next-transpile-modules - an error happened when trying to get the root directory of "${module}". Is it missing a package.json?\n${err}`
+          );
+        }
+        packageRootDirectory = path.dirname(pkgPath);
+      } catch (err) {
+        throw new Error(
+          `next-transpile-modules - an unexpected error happened when trying to resolve "${module}"\n${err}`
+        );
+      }
+
+      return packageRootDirectory;
+    };
```

Comment:
Is there any particular reason why you moved these lines inside the withTM creator? Rather than keeping them outside?

Reply:
The only reason and change is the symlonks option to enhanced-resolve. I passed the resolveSymlinks value from plugin options, so it needed to be inside of it.